diff --git a/libfprint/fp-device.h b/libfprint/fp-device.h
index 9dda747..2466dc3 100644
--- a/libfprint/fp-device.h
+++ b/libfprint/fp-device.h
@@ -122,6 +122,7 @@ typedef enum {
   FP_DEVICE_RETRY_TOO_SHORT,
   FP_DEVICE_RETRY_CENTER_FINGER,
   FP_DEVICE_RETRY_REMOVE_FINGER,
+  FP_DEVICE_RETRY_TOO_FAST,
 } FpDeviceRetry;
 
 /**
diff --git a/libfprint/fprint-list-udev-hwdb.c b/libfprint/fprint-list-udev-hwdb.c
index b726b67..9e5680b 100644
--- a/libfprint/fprint-list-udev-hwdb.c
+++ b/libfprint/fprint-list-udev-hwdb.c
@@ -117,7 +117,6 @@ static const FpIdEntry whitelist_id_table[] = {
   { .vid = 0x27c6, .pid = 0x55a4 },
   { .vid = 0x27c6, .pid = 0x55b4 },
   { .vid = 0x27c6, .pid = 0x5740 },
-  { .vid = 0x27c6, .pid = 0x5e0a },
   { .vid = 0x27c6, .pid = 0x581a },
   { .vid = 0x2808, .pid = 0x9338 },
   { .vid = 0x2808, .pid = 0x93a9 },
diff --git a/libfprint/meson.build b/libfprint/meson.build
index 6df412a..876e0ce 100644
--- a/libfprint/meson.build
+++ b/libfprint/meson.build
@@ -141,6 +141,8 @@ driver_sources = {
         [ 'drivers/goodixmoc/goodix.c', 'drivers/goodixmoc/goodix_proto.c' ],
     'goodixtls511' :
         [ 'drivers/goodixtls/goodix511.c' ],
+    'goodixtls5e0a' :
+        [ 'drivers/goodixtls/goodix5e0a.c', 'drivers/goodixtls/goodix_milan.c' ],
     'fpcmoc' :
         [ 'drivers/fpcmoc/fpc.c' ],
 }
diff --git a/meson.build b/meson.build
index fa27507..fe3ffba 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('libfprint', [ 'c', 'cpp' ],
-    version: '1.94.5',
+    version: '1.94.9',
     license: 'LGPLv2.1+',
     default_options: [
         'buildtype=debugoptimized',
@@ -125,6 +125,7 @@ default_drivers = [
     'upekts',
     'goodixmoc',
     'goodixtls511',
+    'goodixtls5e0a',
     'nb1010',
     'fpcmoc',
 
@@ -160,6 +161,7 @@ driver_helper_mapping = {
     'uru4000' : [ 'nss' ],
     'elanspi' : [ 'udev' ],
     'goodixtls511' : [ 'goodixtls' ],
+    'goodixtls5e0a' : [ 'goodixtls' ],
     'virtual_image'          : [ 'virtual' ],
     'virtual_device'         : [ 'virtual' ],
     'virtual_device_storage' : [ 'virtual' ],
